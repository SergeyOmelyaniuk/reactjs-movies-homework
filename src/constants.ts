import { Language } from './types';

export const languages: Language[] = [
	{ title: 'en', value: 'en-US' },
	{ title: 'ru', value: 'ru-RU' },
];

export const translate: any = {
	en: {
		title: 'title',
		placeholder: 'Movies, person, movie theaters',
		categories: [
			{ id: 1, value: 'popular', title: 'Popular' },
			{ id: 2, value: 'top_rated', title: 'Top rated' },
			{ id: 3, value: 'upcoming', title: 'Upcoming' },
		],
	},
	ru: {
		title: 'заголовок',
		placeholder: 'Фильмы, люди, кинотеатры',
		categories: [
			{ id: 1, value: 'popular', title: 'Популярные' },
			{ id: 2, value: 'top_rated', title: 'Топ рейтинга' },
			{ id: 3, value: 'upcoming', title: 'Ближайшие' },
		],
	},
};
