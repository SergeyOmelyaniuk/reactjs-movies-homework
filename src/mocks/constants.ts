import { Category, Genre, Language } from '../types';

export const categories: Category[] = [
	{ id: 1, value: 'popular', title: 'Popular' },
	{ id: 2, value: 'top_rated', title: 'Top rated' },
	{ id: 3, value: 'upcoming', title: 'Upcoming' },
];

export const languages: Language[] = [
	{ title: 'en', value: 'en-US' },
	{ title: 'ru', value: 'ru-RU' },
];

export const genres: Genre[] = [
	{ id: 1, name: 'Genre 1' },
	{ id: 2, name: 'Genre 2' },
	{ id: 3, name: 'Genre 3' },
	{ id: 4, name: 'Genre 4' },
	{ id: 5, name: 'Genre 5' },
];
