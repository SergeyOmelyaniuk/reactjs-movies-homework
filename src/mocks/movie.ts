import { MovieDetails } from './../types';

export const movieDescription: MovieDetails = {
	poster_path: 'film_1.jpg',
	genres: [{ id: 1, name: 'first genre' }],
	title: 'Красотка на взводе',
	release_date: '2021-05-26',
	vote_average: 68,
	revenue: 120000,
	runtime: 120,
	overview:
		'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
};
