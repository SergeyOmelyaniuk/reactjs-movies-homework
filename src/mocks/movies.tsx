import { Movie } from '../types';

export const movies: Movie[] = [
	{
		id: 1,
		title: 'Красотка на взводе',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_1.jpg',
		vote_average: 68,
		popularity: 0.5,
		revenue: 120000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [
			{
				name: 'Kate Beckinsale',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_1.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Lindy',
			},
			{
				name: 'Stanley Tucci',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_2.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Dr. Ivan Munchin',
			},
			{
				name: 'Bobby Cannavale',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_3.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Detective Vicars',
			},
			{
				name: 'Jai Courtney',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_4.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Justin',
			},
			{
				name: 'Laverne Cox',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_5.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Detective Nevin',
			},
			{
				name: 'David Bradley',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_6.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Gareth Fizel',
			},
		],
	},
	{
		id: 2,
		title: 'Сокровище Счастливчика',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_2.jpg',
		vote_average: 72,
		popularity: 0.5,
		revenue: 230000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 3,
		title: 'Сталинград',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_3.jpg',
		vote_average: 76,
		popularity: 0.5,
		revenue: 240000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 4,
		title: 'Люди в черном: Интернэшнл',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_4.jpg',
		vote_average: 69,
		popularity: 0.5,
		revenue: 136000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 5,
		title: 'Значит, война',
		release_date: '2021-05-26',
		genre: ['Боевик'],
		img: 'film_5.jpg',
		vote_average: 96,
		popularity: 0.5,
		revenue: 226000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
];
