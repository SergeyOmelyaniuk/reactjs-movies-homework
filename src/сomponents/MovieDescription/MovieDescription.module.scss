@import '../../scss/variables.scss';

.movieDescription {
	display: flex;
	@media (max-width: 800px) {
		display: block;
	}
}

.wrapPhotoMovie {
	position: relative;
	max-width: 425px;
	width: 100%;
	height: 640px;
	margin-right: 40px;
	flex-shrink: 0;
	@media (max-width: 1600px) {
		max-width: 352px;
		height: 531px;
		margin-right: 33px;
	}
	@media (max-width: 800px) {
		margin: 0 auto;
	}
}

.photoMovie {
	height: 100%;
	max-width: 100%;
	display: block;
	border-radius: 5px;
}

.infoMovie {
	font-weight: 300;
	font-size: 16px;
	@media (max-width: 1600px) {
		font-size: 13px;
	}
}

.nameMovie {
	font-size: 46px;
	margin-bottom: 50px;
	@media (max-width: 1600px) {
		font-size: 38px;
		margin-bottom: 41px;
	}
}

.wrapGenre {
	margin-bottom: 40px;
	display: flex;
	flex-wrap: wrap;
	@media (max-width: 1600px) {
		margin-bottom: 31px;
	}
}

.genre {
	background-color: $colorGreen;
	padding: 8px 13px;
	border-radius: 5px;
	margin-right: 13px;
	margin-bottom: 10px;
	@media (max-width: 1600px) {
		padding: 7px 11px;
		margin-right: 11px;
	}
}

.wrapTopBilledCast {
	margin-bottom: 60px;
	@media (max-width: 1600px) {
		margin-bottom: 50px;
	}
}

.blockImages {
	margin-bottom: 80px;
	@media (max-width: 1600px) {
		margin-bottom: 66px;
	}
}

.titleImages {
	font-size: 28px;
	display: block;
	margin-bottom: 25px;
	@media (max-width: 1600px) {
		font-size: 23px;
		margin-bottom: 20px;
	}
}

.wrapImages {
	display: flex;
	flex-wrap: wrap;
}

.wrapImage {
	max-width: 250px;
	width: 100%;
	height: 140px;
	margin-bottom: 35px;
	margin-right: 40px;
	&:nth-of-type(n + 5) {
		margin-bottom: 0px;
	}
	&:nth-of-type(4n) {
		margin-right: 0px;
	}
	@media (max-width: 1600px) {
		max-width: 207px;
		height: 116px;
		margin-bottom: 30px;
		margin-right: 33px;
	}
	@media (max-width: 1380px) {
		margin-right: 23px;
		&:nth-of-type(n + 5) {
			margin-bottom: 30px;
		}
		&:nth-of-type(4n) {
			margin-right: 23px;
		}
	}
}

.image {
	height: 100%;
	max-width: 100%;
	display: block;
	border-radius: 5px;
}

.rating {
	display: flex;
	justify-content: center;
	align-items: center;
	position: absolute;
	left: -8px;
	top: 6px;
	background-color: $colorGreen;
	color: $colorWhite;
	width: 56px;
	height: 32px;
	border-radius: 3px;
	font-size: 16px;
	@media (max-width: 1600px) {
		left: -7px;
		top: 5px;
		width: 46px;
		height: 26px;
		font-size: 13px;
	}
}
