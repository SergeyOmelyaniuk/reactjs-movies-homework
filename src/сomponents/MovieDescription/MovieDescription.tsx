import React from 'react';
import TopBilledCast from '../TopBilledCast';
import styles from './MovieDescription.module.scss';

import { Actor, MovieDetails, MovieImages, Language, Genre } from '../../types';
import TextBlock from '../TextBlock';
import { translate } from '../../constants';

interface MovieDescriptionProps {
	movieDetails: MovieDetails;
	movieImages: MovieImages[];
	topCast: Actor[];
	currentLanguage: Language;
}
const numberImagesPerPage = 8;

const MovieDescription = ({
	movieDetails,
	movieImages,
	topCast,
	currentLanguage,
}: MovieDescriptionProps) => {
	const listImages = movieImages.slice(0, numberImagesPerPage);

	return (
		<div className={styles.movieDescription}>
			<div className={styles.wrapPhotoMovie}>
				<img
					className={styles.photoMovie}
					src={`https://image.tmdb.org/t/p/w500/${movieDetails.poster_path}`}
					alt=''
				/>
				<span className={styles.rating}>{movieDetails.vote_average}</span>
			</div>
			<div className={styles.infoMovie}>
				<span className={styles.title}>
					{`${translate[currentLanguage.title].titleMovie}:`}
				</span>
				<h2 className={styles.nameMovie}>{movieDetails.title}</h2>
				<TextBlock
					title={translate[currentLanguage.title].overviewMovie}
					value={movieDetails.overview}
				/>
				<TextBlock
					title={translate[currentLanguage.title].releaseDate}
					value={movieDetails.release_date}
				/>
				<TextBlock
					title={translate[currentLanguage.title].revenue}
					value={`$${movieDetails.revenue}`}
				/>
				<TextBlock
					title={translate[currentLanguage.title].duration}
					value={`${movieDetails.runtime} ${
						translate[currentLanguage.title].minutes
					}`}
				/>
				<div className={styles.wrapGenre}>
					{movieDetails.genres.map((item: Genre) => (
						<div key={item.id} className={styles.genre}>
							{item.name}
						</div>
					))}
				</div>
				<div className={styles.wrapTopBilledCast}>
					<TopBilledCast actors={topCast} currentLanguage={currentLanguage} />
				</div>
				<div className={styles.blockImages}>
					<span className={styles.titleImages}>
						{translate[currentLanguage.title].images}
					</span>
					<div className={styles.wrapImages}>
						{listImages.map((image: MovieImages) => {
							return (
								<div key={image.file_path} className={styles.wrapImage}>
									<img
										className={styles.image}
										src={`https://image.tmdb.org/t/p/w500/${image.file_path}`}
										alt=''
									/>
								</div>
							);
						})}
					</div>
				</div>
			</div>
		</div>
	);
};

export default MovieDescription;
