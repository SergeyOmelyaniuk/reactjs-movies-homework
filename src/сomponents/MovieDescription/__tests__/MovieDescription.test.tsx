import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import MovieDescription from '..';
import { movies } from '../../../mocks/movies';
import { movieDescription } from '../../../mocks/movie';
import { languages } from '../../../constants';

const images = [{ file_path: 'path1' }, { file_path: 'path2' }];

describe('MovieDescription', () => {
	test('should render', () => {
		const { container } = render(
			<BrowserRouter>
				<MovieDescription
					movieDetails={movieDescription}
					movieImages={images}
					topCast={movies.popular[0].actors}
					currentLanguage={languages[0]}
				/>
			</BrowserRouter>
		);

		expect(container).toMatchSnapshot();
	});
});
