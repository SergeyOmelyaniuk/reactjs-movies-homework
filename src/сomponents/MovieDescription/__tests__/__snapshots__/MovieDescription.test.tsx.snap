// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDescription should render 1`] = `
<div>
  <div
    class="movieDescription"
  >
    <div
      class="wrapPhotoMovie"
    >
      <img
        alt="movie"
        class="photoMovie"
        src="https://image.tmdb.org/t/p/w500/film_1.jpg"
      />
      <span
        class="rating"
      >
        68
      </span>
    </div>
    <div
      class="infoMovie"
    >
      <span
        class="title"
      >
        Title:
      </span>
      <h2
        class="nameMovie"
      >
        Красотка на взводе
      </h2>
      <div
        class="textBlock"
      >
        <span
          class="title"
        >
          Overview
          :
        </span>
        <p
          class="text"
        >
          Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.
        </p>
      </div>
      <div
        class="textBlock"
      >
        <span
          class="title"
        >
          Release date
          :
        </span>
        <p
          class="text"
        >
          2021-05-26
        </p>
      </div>
      <div
        class="textBlock"
      >
        <span
          class="title"
        >
          Revenue
          :
        </span>
        <p
          class="text"
        >
          $120000
        </p>
      </div>
      <div
        class="textBlock"
      >
        <span
          class="title"
        >
          Duration
          :
        </span>
        <p
          class="text"
        >
          120 minutes
        </p>
      </div>
      <div
        class="wrapGenre"
      >
        <div
          class="genre"
        >
          first genre
        </div>
      </div>
      <div
        class="wrapTopBilledCast"
      >
        <div
          class="topBilledCast"
        >
          <div
            class="header"
          >
            <span
              class="title"
            >
              Top billed cast
            </span>
            <button
              class="button"
            >
              Show all
            </button>
          </div>
          <div
            class="wrap"
          >
            <a
              class="actorCard"
              href="/actors/1"
            >
              <div
                class="wrapImage"
              >
                <img
                  alt="actor"
                  class="image"
                />
              </div>
              <div
                class="content"
              >
                <h2
                  class="title"
                >
                  Kate Beckinsale
                </h2>
                <p
                  class="character"
                >
                  Lindy
                </p>
              </div>
            </a>
            <a
              class="actorCard"
              href="/actors/2"
            >
              <div
                class="wrapImage"
              >
                <img
                  alt="actor"
                  class="image"
                />
              </div>
              <div
                class="content"
              >
                <h2
                  class="title"
                >
                  Stanley Tucci
                </h2>
                <p
                  class="character"
                >
                  Dr. Ivan Munchin
                </p>
              </div>
            </a>
            <a
              class="actorCard"
              href="/actors/3"
            >
              <div
                class="wrapImage"
              >
                <img
                  alt="actor"
                  class="image"
                />
              </div>
              <div
                class="content"
              >
                <h2
                  class="title"
                >
                  Bobby Cannavale
                </h2>
                <p
                  class="character"
                >
                  Detective Vicars
                </p>
              </div>
            </a>
            <a
              class="actorCard"
              href="/actors/4"
            >
              <div
                class="wrapImage"
              >
                <img
                  alt="actor"
                  class="image"
                />
              </div>
              <div
                class="content"
              >
                <h2
                  class="title"
                >
                  Jai Courtney
                </h2>
                <p
                  class="character"
                >
                  Justin
                </p>
              </div>
            </a>
            <a
              class="actorCard"
              href="/actors/5"
            >
              <div
                class="wrapImage"
              >
                <img
                  alt="actor"
                  class="image"
                />
              </div>
              <div
                class="content"
              >
                <h2
                  class="title"
                >
                  Laverne Cox
                </h2>
                <p
                  class="character"
                >
                  Detective Nevin
                </p>
              </div>
            </a>
            <a
              class="actorCard"
              href="/actors/6"
            >
              <div
                class="wrapImage"
              >
                <img
                  alt="actor"
                  class="image"
                />
              </div>
              <div
                class="content"
              >
                <h2
                  class="title"
                >
                  David Bradley
                </h2>
                <p
                  class="character"
                >
                  Gareth Fizel
                </p>
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        class="blockImages"
      >
        <span
          class="titleImages"
        >
          Images
        </span>
        <div
          class="wrapImages"
        >
          <div
            class="wrapImage"
          >
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/path1"
            />
          </div>
          <div
            class="wrapImage"
          >
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/path2"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
